docs/metrics.md

```diff
@@ -0,0 +1,561 @@
+
+In real world applications, the cardinality can be extremely high. Imagine if we
+have a long running service and we collect metrics with 7 attributes and each
+attribute can have 30 different values. We might eventually end up having to
+remember the complete set of all 21,870,000,000 combinations! This cardinality
+explosion is a well-known challenge in the metrics space. For example, it can
+cause surprisingly high costs in the observability system, or even be leveraged
+by hackers to launch a denial-of-service attack.
+
+[Cardinality
+limit](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#cardinality-limits)
+is a throttling mechanism which allows the metrics collection system to have a
+predictable and reliable behavior when excessive cardinality happens, whether it
+was due to a malicious attack or developer making mistakes while writing code.
+
+OpenTelemetry has a default cardinality limit of `2000` per metric. This limit
+can be configured at the individual metric level using the [View
+API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view)
+leveraging the
+[`cardinality_limit`](https://docs.rs/opentelemetry_sdk/latest/opentelemetry_sdk/metrics/struct.Stream.html#structfield.cardinality_limit)
+setting.
+
+It's important to understand that this cardinality limit applies only at the
+OpenTelemetry SDK level, not to the ultimate cardinality of the metric as seen
+by the backend system. For example, while a single process might be limited to
+2000 attribute combinations per metric, the actual backend metrics system might
+see much higher cardinality due to:
+
+1. Resource attributes (such as `service.instance.id`, `host.name`, etc.) that
+   can be added to each metric
+2. Multiple process instances running the same application across your
+   infrastructure
+3. The possibility of reporting different key-value pair combinations in each
+   export interval, as the cardinality limit only applies to the number of
+   unique attribute combinations tracked during a single export interval. (This
+   is only applicable to Delta temporality)
+
+Therefore, the actual cardinality in your metrics backend can be orders of
+magnitude higher than what any single OTel SDK process handles in an export
+cycle.
+
+#### Cardinality Limit - Implications
+
+Cardinality limits are enforced during each export interval, meaning the metrics
+aggregation system only allows up to the configured cardinality limit of unique
+attribute combinations per metric. Understanding how this works in practice is
+important:
+
+* **Cardinality Capping**: When the limit is reached within an export interval,
+  any new attribute combination is not individually tracked but instead folded
+  into a single aggregation with the attribute `{"otel.metric.overflow": true}`.
+  This ensures that measurements are never lost, even when cardinality explodes.
```

Comment:
This sentence is a bit misleading as it might be misconstrued as there is no data loss. We completely lose the attributes information of the measurement.

```suggestion
  This ensures that we correctly retain the high-level aggregates for the metric such as Sum and Count. This information can then be used to make informed decisions about the required cardinality for the metric.
```